### ISSUE TITLE
feat(linter): bad-bitwise-operator rule

### ISSUE THREAD

**Issue by mysteryven** [Boshen/oxc#243]
Implement Deepscan's [bad-bitwise-operator](https://deepscan.io/docs/rules/bad-bitwise-operator) rule.

### Question

This rule has different result depending on the type of variable.

```js
var a = 1
input |= a // pass
a = "1"
input |= a // error

var defaultVal = 1
a = options | defaultVal // pass
defaultVal = "1"
a = options | defaultVal  // error
```

Is there some way to get the identifier's current type in Oxc? 🧐

**Comment by Boshen** [comment]
We can only check for operands that are definitely not a number, we don't have type inference for now.

Please note, this rule is safe to be a correctness rule because the types are wrong. The eslint `no-bitwise` rule cannot be correctness because it doesn't check types.

**Comment by Boshen** [comment]
In the vscode repo, there are lots of false positives on enum bitflags.

So we need to decide on demoting the rule to another category or run this on js files only before closing this issue.

**Comment by mysteryven** [comment]
I apologize for the question I have to ask: How to run the unpublished package in the VSCode repo?

Here are the steps I tried: 
1. Run `cargo build --release --target ${{ matrix.target }} -p oxc_cli` 
2. Move the binary executable oxc_cli to a JS repo (node_modules/@oxidation-compiler/darwin-arm64/oxc)
3. Run it, and it throws error below

```bash
<jemalloc>: Error allocating TSD
fish: Job 1, './oxc' terminated by signal SIGABRT (Abort)
```

4. I add this rule to my global bin path, it also throws the same error.

**Comment by mysteryven** [comment]
Please ignore, I solved it.😅

**Comment by Boshen** [comment]
You can find the binaries in ./target/release and ./target/debug

**Comment by Boshen** [comment]
Moved to restriction because this should be opt-in, it doesn't matter if it's js or ts.
